### PR TITLE
IfPipe can lose the input message

### LINKED ISSUE #9012: ifPipe loses the input
### Current Behavior

When using the ifpipe in combination with JSON payload (did not test with xml) the pipes input is lost

​
```
​
<SenderPipe name="Get temperature from OpenWeather"  storeResultInSessionKey="weather">
             <HttpSender url="http://api.openweathermap.org/data/2.5/weather">
                <Param name="q" value="Amsterdam"/>
                <Param name="units" value="metric"/>
                <Param name="mode" value="json"/>
                <Param name="appid" value="xxxxxxxxxxxxxxxxxxxxxxxx"/>
             </HttpSender>
          </SenderPipe>
          <IfPipe name="Temperature present?" jsonPathExpression=" $.main[?(@.temp)].temp"  defaultMediaType="JSON">
             <Forward name="then" path="Map response" />
             <Forward name="else" path="Return error message" />
          </IfPipe>
          <EchoPipe
             name="Return error message"
             getInputFromFixedValue="Temperature not found in response of api.openweathermap.org"
             >
             <Forward name="error" path="InternalServerError" />
          </EchoPipe>
          <EchoPipe
             name="Map response">
          </EchoPipe>

```

preserve input does not help

### Expected Behavior

the output of the if pipe should be the input 

### Environment Information

9.2 local frankrunner

### Steps To Reproduce

_No response_

### Configuration

```xml

```

### Input

_No response_

### What database are you using?

_No response_

### What browsers are you seeing the problem on?

_No response_

### Relevant Log Output

```console

```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tnleeuw**
Closes #9012: IfPipe is supposed to pass on the input unmodified but when the input is a stream, there are various ways in which it can lose the input.

### CODE REVIEW

File: core/src/main/java/org/frankframework/util/MessageUtils.java [hunk 1]
```diff
@@ -332,9 +333,15 @@ private static MimeType guessMimeType(Message message) {
 		if (!"{".equals(firstChar) && !"[".equals(firstChar)) {
 			return MediaType.TEXT_PLAIN;
 		}
-		try (JsonParser parser = Json.createParser(message.asInputStream())) {
-			parser.next();
-			return MediaType.APPLICATION_JSON;
+		try {
+			InputStream inputStream = message.asInputStream();
+			inputStream.mark(20_000);
```
Comment (nielsm5) [inline core/src/main/java/org/frankframework/util/MessageUtils.java:338]: Ik weet niet of de jsonParser meer wilt inlezen en dit nu goed gaat omdat onze test input niet groter is dan 20_000L

Comment (tnleeuw) [reply core/src/main/java/org/frankframework/util/MessageUtils.java:338]: Dat risico is er. 20K leek me een redelijke buffer.
Ik kan dit hele stuk er ook uithalen en maar aannemen dat het JSON zal zijn als het begint met `{` of `[`, daar heb ik over getwijfeld.

**Review by evandongen — APPROVED** [review]

File: core/src/main/java/org/frankframework/util/MessageUtils.java (hunk 1 above)
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/util/MessageUtils.java:338]: PS: De parser is een streaming parser. Hij zal niet meteen de hele JSON inlezen, behalve misschien in bepaalde edge-cases.
Daarom verwacht ik niet direct een probleem.

Comment (tnleeuw) [reply core/src/main/java/org/frankframework/util/MessageUtils.java:338]: Ik heb 2 tests toegevoegd voor grote JSON berichten. Lijkt te werken, in ieder geval met deze berichten.
